File: scratch/create_wallet.go
//go:build ignore

package main

import (
	"crypto/ed25519"
	"crypto/rand"
	"encoding/hex"
	"fmt"
	"log"
	"os"
	"synthos-collective/internal/chain"
)

func main() {
	pub, priv, err := ed25519.GenerateKey(rand.Reader)
	if err != nil {
		log.Fatalf("Failed to generate key: %v", err)
	}

	addr := chain.AddressFromPublicKey(pub)
	pubHex := "0x" + hex.EncodeToString(pub)
	privHex := "0x" + hex.EncodeToString(priv)

	fmt.Printf("Wallet Address: %s\n", addr)
	fmt.Printf("Public Key:     %s\n", pubHex)
	fmt.Println("Private Key:    <written to scratch/USER_WALLET.local.txt>")

	walletInfo := fmt.Sprintf("Address: %s\nPublic Key: %s\nPrivate Key: %s\n", addr, pubHex, privHex)
	err = os.WriteFile("scratch/USER_WALLET.local.txt", []byte(walletInfo), 0600)
	if err != nil {
		log.Fatalf("Failed to save wallet: %v", err)
	}
}

File: scratch/create_wallet.js
const crypto = require('crypto');
const fs = require('fs');

const { publicKey, privateKey } = crypto.generateKeyPairSync('ed25519');

const pubHex = "0x" + publicKey.export({ type: 'spki', format: 'der' }).toString('hex').slice(-64);
const privHex = "0x" + privateKey.export({ type: 'pkcs8', format: 'der' }).toString('hex').slice(-64);

// Address derivation in Synthos seems to be based on public key.
// Looking at internal/chain/address.go or similar.
// In cmd/node/main.go: addr := chain.AddressFromPublicKey(pub)

// Let's just output the keys and I'll derive the address if I can see the code.
const output = `Public Key: ${pubHex}\nPrivate Key: ${privHex}\n`;
fs.writeFileSync('scratch/USER_WALLET_NODE.local.txt', output, { mode: 0o600 });
console.log(`Public Key: ${pubHex}`);
console.log('Private Key: <written to scratch/USER_WALLET_NODE.local.txt>');
